parachain/pallets/inbound-queue/src/lib.rs

```diff
@@ -330,6 +336,33 @@ pub mod pallet {
 				.saturating_add(len_fee)
 				.saturating_add(T::PricingParameters::get().rewards.local)
 		}
+
+		/// Burn the amount of the fee embedded into the XCM for teleports
+		pub fn burn_fees(para_id: ParaId, fee: BalanceOf<T>) -> DispatchResult {
+			let dummy_context =
+				XcmContext { origin: None, message_id: Default::default(), topic: None };
+
+			let dest = MultiLocation { parents: 1, interior: X1(Parachain(para_id.into())) };
+
+			let fees = (MultiLocation::parent(), fee.saturated_into::<u128>()).into();
+
+			<T::XcmExecutor as XcmAssetTransfers>::AssetTransactor::can_check_out(
+				&dest,
+				&fees,
+				&dummy_context,
+			)
+			.map_err(|_| Error::<T>::BurnFees)?;
+
+			<T::XcmExecutor as XcmAssetTransfers>::AssetTransactor::check_out(
+				&dest,
+				&fees,
+				&dummy_context,
+			);
+
+			let sovereign_account = sibling_sovereign_account::<T>(para_id);
+			T::Token::burn_from(&sovereign_account, fee, Precision::Exact, Fortitude::Polite)?;
```

Comment:
For consistency, I think we should use `T::AssetTransactor::withdraw_asset`. Thoughts?

Reply:
Nice!